packages/test-utils/src/structAbiEncoder.ts

```diff
@@ -0,0 +1,33 @@
+import * as ethers from "ethers";
+
+export class StructAbiEncoder {
+  public static fromDefinition(solidityDefinition: string): StructAbiEncoder {
+    const definitions: string[] = [];
+    const lines = solidityDefinition.split(";");
+    for (const line of lines) {
+      const definition = line.trim();
+      if (definition.length === 0) {
+        continue;
+      }
+      const parts = definition.split(" ");
+      if (parts.length !== 2) {
+        throw new Error(
+          `Invalid struct field. Expected '[type] [name]', got '${definition}'`
+        );
+      }
+      definitions.push(definition);
+    }
+    const encoding = `tuple(${definitions.join(", ")})`;
```

Comment:
It would be cool to extend this in the future to support encodings with nested tuples.

Reply:
Yeah, shouldn't be hard to do